charts/harvester-cloud-provider/templates/hardened.yaml

```diff
@@ -0,0 +1,119 @@
+{{- $namespaces := dict "_0" .Release.Namespace -}}
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ .Chart.Name }}-patch-sa
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: {{ .Chart.Name }}-patch-sa
+  annotations:
+    "helm.sh/hook": post-install, post-upgrade
+    "helm.sh/hook-delete-policy": hook-succeeded, before-hook-creation
+spec:
+  template:
+    metadata:
+      name: {{ .Chart.Name }}-patch-sa
+      labels:
+        app: {{ .Chart.Name }}-patch-sa
+    spec:
+      serviceAccountName: {{ .Chart.Name }}-patch-sa
+      securityContext:
+        runAsNonRoot: true
+        runAsUser: 1000
+      restartPolicy: Never
+      containers:
+        {{- range $_, $ns := $namespaces }}
+        - name: patch-sa-{{ $ns }}
+          image: {{ template "system_default_registry" $ }}{{ $.Values.global.kubectl.repository }}:{{ $.Values.global.kubectl.tag }}
+          imagePullPolicy: {{ $.Values.global.kubectl.pullPolicy }}
+          command: ["kubectl", "patch", "serviceaccount", "default", "-p", "{\"automountServiceAccountToken\": false}"]
+          args: ["-n", "{{ $ns }}"]
+  {{- end }}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: {{ .Chart.Name }}-patch-sa
+  labels:
+    app: {{ .Chart.Name }}-patch-sa
+rules:
+  - apiGroups:
+      - ""
+    resources:
+      - serviceaccounts
+    verbs: ['get', 'patch']
+  - apiGroups: ['policy']
+    resources: ['podsecuritypolicies']
+    verbs:     ['use']
+    resourceNames:
+      - {{ .Chart.Name }}-patch-sa
+---
+apiVersion: rbac.authorization.k8s.io/v1
```

Comment:
adding this isn't sufficient for hardened cluster support, the default deployment also needs a PSP that it can use.

So the ClusterRole in https://github.com/harvester/charts/blob/57f5c07d6b1eb1c47b25b5e55a4362a7e9eb649f/charts/harvester-cloud-provider/templates/rbac.yaml#L14-L22 needs to be able to use a PSP and that PSP needs to also be defined.

Reply:
I wonder why the default deployment needs a PSP that it can use. @aiyengar2

Reply:
As listed in https://kubernetes.io/docs/concepts/policy/pod-security-policy/#enabling-pod-security-policies, `PodSecurityPolicies are enforced by enabling the admission controller, but doing so without authorizing any policies will prevent any pods from being created in the cluster.`.

Therefore, every deployment, job, etc. that creates a pod needs to have a PodSecurityPolicy defined that it has permissions to use.